command/agent/testdata/basic.json

```diff
@@ -76,6 +76,7 @@
       "alloc_mounts_dir": "/tmp/mounts",
       "bridge_network_name": "custom_bridge_name",
       "bridge_network_subnet": "custom_bridge_subnet",
+      "bridge_network_subnet_ipv6": "custom_bridge_subnet_ipv6",
```

Comment:
It's a little weird that our tests have invalid values here (I think these aren't valid?)

Reply:
It's a good point - they're not validated anywhere until they actually get used at allocation-time.

I always hated finding out about bad client config that way, when an alloc fails through no fault of the jobspec.  I would prefer to have the client fail to start instead of be sitting around quietly poisoned.  Would this be a good place to add a check?

https://github.com/hashicorp/nomad/pull/23882/files#diff-430a3b367fc84668c6ae1cb6936b7a7b5c5b321f42aa754c23618f848faf87b1R893

I can also add validation for custom ipv4 subnet while I'm in here, too. A major version bump is probably a good time to go for that, eh?